api/query.go

```diff
@@ -241,7 +243,7 @@ func (q *QueryTableResult) Next() bool {
 readRow:
 	row, q.err = q.csvReader.Read()
 	if q.err == io.EOF {
-		q.err = nil
+		q.err = q.Close()
```

Comment:
Actually, I have a question here. Shouldn't `q.Close()` be called also on line 250?